shared/query.go

```diff
@@ -95,27 +207,69 @@ func QueryImage(ctx context.Context, imgUrl *url.URL, w io.Writer) (err error) {
 
 	slog.InfoContext(ctx, "querying image", "url", imgUrl.String())
 
-	req, err := makeRequest(ctx, imgUrl)
-	if err != nil {
-		return err
-	}
+	entry, err := cache.Get(imgUrl.String())
+
+	if err == nil {
+		slog.InfoContext(ctx, "loading cache of image", "url", imgUrl.String())
+	} else {
+		var req *http.Request
+		var res *http.Response
 
-	req.Header.Add("Accept", mime.TypeByExtension(".png"))
-	req.Header.Add("Accept", mime.TypeByExtension(".jpg"))
+		for i := 0; i < 3; i++ {
+			if res != nil {
+				_ = res.Body.Close()
+			}
 
-	res, err := http.DefaultClient.Do(req)
-	if err != nil {
-		return err
-	}
-	defer res.Body.Close()
+			err = QueryImageLimiter.Wait(ctx)
```

Comment:
Same as above, this section can be moved into a separate function. Perhaps the two can be combined?